com.unity.render-pipelines.core/Runtime/Debugging/VolumeDebugSettings.cs

```diff
@@ -16,41 +16,46 @@ public abstract partial class VolumeDebugSettings<T> : IVolumeDebugSettings2
         /// <summary>Current volume component to debug.</summary>
         public int selectedComponent { get; set; } = 0;
 
-        private Camera m_SelectedCamera;
-
         /// <summary>Current camera to debug.</summary>
-        public Camera selectedCamera
-        {
-            get => m_SelectedCamera ?? cameras.FirstOrDefault();
-            set => m_SelectedCamera = value;
-        }
+        public Camera selectedCamera { get; set; }
+
+        private readonly List<Camera> m_Cameras = new List<Camera>();
+        private Camera[] m_CamerasArray;
 
         /// <summary>Returns the collection of registered cameras.</summary>
         public IEnumerable<Camera> cameras
         {
             get
             {
+                m_Cameras.Clear();
 
 #if UNITY_EDITOR
                 if (SceneView.lastActiveSceneView != null)
                 {
                     var sceneCamera = SceneView.lastActiveSceneView.camera;
                     if (sceneCamera != null)
-                        yield return sceneCamera;
+                    {
+                        m_Cameras.Add(sceneCamera);
+                    }
                 }
 #endif
+
+                if (m_CamerasArray == null || Camera.allCamerasCount != m_CamerasArray.Length)
+                {
+                    m_CamerasArray = new Camera[Camera.allCamerasCount];
+                    Camera.GetAllCameras(m_CamerasArray);
```

Comment:
What's the purpose of getting the cameras to m_CamerasArray if you don't the array afterwards? Should line 49 be iterating over m_CamerasArray instead of Camera.allCameras?

Reply:
Yeah, sorry! I forgot to commit the files, my bad! Nice catch btw :)